Add mandoc output support for manpages

ngie-eign opened the pull request:
This change modifies the Makefile generator to support mandoc format
manpages, in lieu of \*roff format manpages.
    
After this commit the user has the ability of specifying the manpage
format to the `--manpage-format` flag. The 2 supported manpage formats
as of writing are "mdoc" and "roff" and the default remains the "roff"    
format for legacy and portability reasons. 
    
The mandoc format requires pod2mdoc to be installed, whereas the roff
output format requires pod2man to be installed. The former requires an
additional utility be installed, whereas the latter uses pod2man, a
utility that has been present with perl distributions for well over a
decade.
        
mandoc format support is being added as it is an easier/arguably more
structured manpage format to parse, making it easier for downstream
consumers like FreeBSD to implement OS-specific build support, as the
minimum dependencies for the OpenSSL build process are more involved
than the tools available for the FreeBSD OS bootstrapping process.

Closes: #28325

ngie-eign committed:
Add mandoc output support for manpages

This change modifies the Makefile generator to support mandoc format
manpages, in lieu of \*roff format manpages.

After this commit the user has the ability of specifying the manpage
format to the `--manpage-format` flag. The 2 supported manpage formats
as of writing are "mdoc" and "roff" and the default remains the "roff"
format for legacy and portability reasons.

The mandoc format requires pod2mdoc to be installed, whereas the roff
output format requires pod2man to be installed. The former requires an
additional utility be installed, whereas the latter uses pod2man, a
utility that has been present with perl distributions for well over a
decade.

mandoc format support is being added as it is an easier/arguably more
structured manpage format to parse, making it easier for downstream
consumers like FreeBSD to implement OS-specific build support, as the
minimum dependencies for the OpenSSL build process are more involved
than the tools available for the FreeBSD OS bootstrapping process.

Signed-off-by: Enji Cooper <yaneurabeya@gmail.com>

levitte commented:
Do you mind putting the INSTALL.md changes in a separate PR?  (and oh, after "Environment variables", there's a "Reconfigure" heading that should become an h2 as well)

openssl-machine commented:
This PR has the label 'hold: cla required' and is stale: it has not been updated in 30 days. Note that this PR may be automatically closed in the future if no CLA is provided.  For CLA help see https://www.openssl.org/policies/cla.html

ngie-eign committed:
INSTALL.md: document `--manpage-format` changes

This change documents the new Configure option, `--manpage-format`.

Address formatting issues and expound on the `Reconfigure` section.

Signed-off-by: Enji Cooper <yaneurabeya@gmail.com>

t8m commented on Configurations/unix-Makefile.tmpl:
```diff
@@ -1583,6 +1583,16 @@ EOF
+		 $pod >\$\@
+EOF
+          }
+          die "Unhandled manpage format: $config{manpage_format}" unless ($config{manpage_format} eq "roff");
```
Should the generation of the roff format below be skipped if the config value is mdoc?

t8m reviewed (changes requested):
This needs documentation in INSTALL.md

t8m commented:
> This needs documentation in INSTALL.md

Ah I see that is in PR https://github.com/openssl/openssl/pull/29020 however I do not understand why this should be in a separate PR as this should go in at once. Could you please just add the commit here and close the other PR?

ngie-eign commented:
> > @levitte asked for it to be a separate PR in an earlier comment: [#28450 (comment)](https://github.com/openssl/openssl/pull/28450#issuecomment-3257125793) . Maybe it was a typo?
> 
> Yes, I saw that but not sure why he asked for a separate PR - this PR is adding the option and the documentation should go with it as the first thing should not be merged without the other thing.

I'm not genuinely sure 🤷...

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 61 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 92 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 123 days ago

ngie-eign commented:
@levitte , @t8m ... Anyone?

beldmit reviewed (approved):
Looks like the changes were implemented.

The changes in INSTALL.md are probably not necessary but I can stand it.

LGTM

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 30 days ago

ngie-eign commented:
> @levitte , @t8m ... Anyone?

@t8m : could you please re-review/possibly merge the change?

ngie-eign commented:
Should this change be documented in `CHANGES.md`, or is this a `no_news_changes_needed` label candidate?

t8m reviewed (approved):
I think it is worth adding a CHANGES.md entry but it can be added after this PR is merged.

openssl-machine commented:
This pull request is ready to merge

jogme commented:
Merged to master. Thank you for the patience!
